src/cache.c

```diff
@@ -115,9 +115,12 @@ void git_cache_dispose(git_cache *cache)
 /* Called with lock */
 static void cache_evict_entries(git_cache *cache)
 {
-	size_t evict_count = 8, i;
+	size_t evict_count = git_cache_size(cache) / 2048, i;
```

Comment:
Yeah, this makes a lot of sense to me. Did you experiment with different divisors here? Just asking out of curiosity as I wonder how you arrived at 2048

Reply:
Yes. I first started off with 16384 and then went down to 8192. With 8192, we were still taking 20% of pack time in `git_oidmap_iterate`. With 2048, we're spending 6.2% there, which is much lower.

I didn't want to go down much more because each eviction can start to get expensive with a 256 MB cache, which means that if we use a medium-sized repository where we only end up evicting once or twice, eviction costs may be much more significant. While I want to improve performance for repositories that are Linux-sized or larger, I also don't want to regress it for smaller repositories.

We also have to consider that as we evict more entries, we do more re-reads from the actual object database, since some of the items we had cached are now gone. 2048 is still an improvement in this area, since total time in `odb_read_1` and its children are down. It's possible that 1024 would also work, but I didn't test it.

I picked a power of two because it's cheaper to compute in a hot path and the exact number doesn't matter too much.

Reply:
Perfect, thanks a lot for explaining!